lib/Syntax/Syntax.cpp

```diff
@@ -81,53 +81,38 @@ llvm::Optional<Syntax> Syntax::getParent() const {
   };
 }
 
+Syntax Syntax::getRoot() const {
+  return { Root, Root.get() };
+}
+
 size_t Syntax::getNumChildren() const {
-  size_t NonTokenChildren = 0;
-  for (auto Child : getRaw()->getLayout()) {
-    if (!Child->isToken()) {
-      ++NonTokenChildren;
-    }
-  }
-  return NonTokenChildren;
+  return Data->getNumChildren();
```

Comment:
I remember writing this and we had some tests that, for some reason, expected it to skip token children. Didn't want to mess with it at the time, because I was doing a behavior-preserving refactor. As far as I'm concerned, it should _not_ skip tokens.

Reply:
Originally, we defined children as something indicating recursive structure in the tree but I agree, this should not skip tokens. This may indicate the name *children* isn't quite right. FWIW, Roslyn likes to use the term *slots*, similar to how we also use the term *layout*. Whether you can descend into a node is irrelevant to the number of items in the syntax's layout.